### PR TITLE
Add benchmark for cluster API

### PR DESCRIPTION
**Pull request by akudiyar**
## Before memory optimizations

Benchmark results on commit https://github.com/tarantool/cartridge-java/commit/a6199e8268577517167492191b0c5989fbcf99e8 (before optimizations):
```
Total time: 00:33:33

Benchmark                                            Mode  Cnt      Score      Error  Units
ClusterBenchmarkRunner.readDataUsingCallAPI         thrpt   10  13761.025 ± 2657.463  ops/s
ClusterBenchmarkRunner.readDataUsingCallAPI:·jfr    thrpt             NaN               ---
ClusterBenchmarkRunner.readDataUsingSpaceAPI        thrpt   10  12033.690 ± 2032.909  ops/s
ClusterBenchmarkRunner.readDataUsingSpaceAPI:·jfr   thrpt             NaN               ---
ClusterBenchmarkRunner.writeDataUsingCallAPI        thrpt   10   6175.897 ±  630.966  ops/s
ClusterBenchmarkRunner.writeDataUsingCallAPI:·jfr   thrpt             NaN               ---
ClusterBenchmarkRunner.writeDataUsingSpaceAPI       thrpt   10   7123.615 ±  894.854  ops/s
ClusterBenchmarkRunner.writeDataUsingSpaceAPI:·jfr  thrpt             NaN               ---
```

Allocations for `ClusterBenchmarkRunner.writeDataUsingCallAPI` (before optimizations):

Allocation_before

TLAB allocations for `ClusterBenchmarkRunner.writeDataUsingCallAPI` (before optimizations):

TLAB_allocation_before

## After memory optimizations

Benchmark results on commit https://github.com/tarantool/cartridge-java/pull/439/commits/5f9e69c8d9d0e907e356bb5954c547364fa5207d (after optimizations):
```
Total time: 00:24:56

Benchmark                                            Mode  Cnt      Score      Error  Units
ClusterBenchmarkRunner.readDataUsingCallAPI         thrpt   10  19674.739 ± 2977.708  ops/s
ClusterBenchmarkRunner.readDataUsingCallAPI:·jfr    thrpt             NaN               ---
ClusterBenchmarkRunner.readDataUsingSpaceAPI        thrpt   10  17092.066 ± 5721.286  ops/s
ClusterBenchmarkRunner.readDataUsingSpaceAPI:·jfr   thrpt             NaN               ---
ClusterBenchmarkRunner.writeDataUsingCallAPI        thrpt   10   8012.627 ± 1120.023  ops/s
ClusterBenchmarkRunner.writeDataUsingCallAPI:·jfr   thrpt             NaN               ---
ClusterBenchmarkRunner.writeDataUsingSpaceAPI       thrpt   10   9157.278 ± 3705.178  ops/s
ClusterBenchmarkRunner.writeDataUsingSpaceAPI:·jfr  thrpt             NaN               ---
```

Allocations for `ClusterBenchmarkRunner.writeDataUsingCallAPI` (after optimizations):

Allocation_after

TLAB allocations for `ClusterBenchmarkRunner.writeDataUsingCallAPI` (after optimizations):

TLAB_allocation_after

## Summary

The average throughput gain is about 40% on read operations and 30% on write operations. The benchmark run completes approx. 30% faster with optimizations. The main reason for improvements can be a >50% reduction in allocations.

Benchmark runs were performed with JDK 17, G1 GC (default settings), using the following command:
```
mvn exec:exec -Pbenchmark -Dbenchmark="ClusterBenchmarkRunner" -DbenchmarkArgs="-prof=jfr"
```

### CODE REVIEW

File: src/main/java/io/tarantool/driver/api/MessagePackMapperBuilder.java [hunk 1]
```diff
@@ -7,6 +7,7 @@
 import org.msgpack.value.Value;
 import org.msgpack.value.ValueType;
 
+import java.util.Collection;
```
Comment (ArtDu) [inline src/main/java/io/tarantool/driver/api/MessagePackMapperBuilder.java:10]: nit: probably unused

File: src/test/java/io/tarantool/driver/benchmark/ClusterTarantoolSetup.java [hunk 2]
```diff
@@ -55,8 +54,8 @@ private void initClient() {
                 new TarantoolServerAddress(tarantoolContainer.getRouterHost(), tarantoolContainer.getMappedPort(3303))
             )
             .withCredentials(tarantoolContainer.getUsername(), tarantoolContainer.getPassword())
-            .withConnections(10)
-            .withEventLoopThreadsNumber(10)
+            .withConnections(2)
+            .withEventLoopThreadsNumber(2)
```
Comment (ArtDu) [inline src/test/java/io/tarantool/driver/benchmark/ClusterTarantoolSetup.java:58]: nit:

I think it's more interesting to compare on 1 connection = 1 thread. I don't think we need more. Throughput will be increased with thread increasing but it depends on how many cores you have on running pc

**Review by ArtDu — APPROVED** [review]
Looks sagnificent. Only nitpick comments

File: src/main/java/io/tarantool/driver/api/MessagePackMapperBuilder.java (hunk 1 above)
Comment (akudiyar) [reply src/main/java/io/tarantool/driver/api/MessagePackMapperBuilder.java:10]: It was added for fixing a JavaDoc warning

File: src/test/java/io/tarantool/driver/benchmark/ClusterTarantoolSetup.java (hunk 2 above)
Comment (akudiyar) [reply src/test/java/io/tarantool/driver/benchmark/ClusterTarantoolSetup.java:58]: Yeah, perhaps, but I also believe that the behavior will be too much like the behavior of internal netty parts, and it is too far from the real-world scenario.

**Pull request merged**